asset/records.go

```diff
@@ -35,9 +35,9 @@ const (
 type WitnessTlvType = tlv.Type
 
 const (
-	WitnessPrevID          WitnessTlvType = 0
-	WitnessTxWitness       WitnessTlvType = 1
-	WitnessSplitCommitment WitnessTlvType = 2
+	WitnessPrevID          WitnessTlvType = 1
+	WitnessTxWitness       WitnessTlvType = 3
+	WitnessSplitCommitment WitnessTlvType = 5
```

Comment:
Same here, both the `PrevID` and `TxWitness` fields have an explicit check in `EncodeRecords`: https://github.com/lightninglabs/taproot-assets/blob/5373ccceaa92a6b4e32ff55c2e272608429fbbc5/asset/asset.go#L340

I guess if we declare `PrevID` as required, it would no longer need to be a pointer? Not sure how to decide what to take as the source of truth for the decision: our current implementation or the fact that it should in practice never be empty...

Reply:
Indeed, its the `ZeroPrevID` in both forms of the genesis witness, and the VM already fails if its nil. In favor of promoting to not-pointer.

TxWitness is fine as-is though IIUC should only be empty for genesis assets.